src/test/Distributor.t.sol

```diff
@@ -0,0 +1,215 @@
+pragma solidity =0.8.11;
+
+import "ds-test/test.sol";
+import "forge-std/Vm.sol";
+
+import "src/Distributor.sol";
+import "src/test/__fixtures/MintableERC20.sol";
+
+contract DistributorTest is DSTest
+{
+    Vm private vm = Vm(HEVM_ADDRESS);
+
+    Distributor private distributor; 
+    MintableERC20 private incomingToken = new MintableERC20("Wrapped Vechain", "WVET");
+    MintableERC20 private tokenToRecover = new MintableERC20("Recover Me", "RME");
+    address private recipient1 = address(1);
+    address private recipient2 = address(2);
+
+    function setUp() public
+    {
+        distributor = new Distributor(incomingToken);
+    }
+
+    function testIncomingToken() public
+    {
+        // assert
+        assertEq(address(distributor.incomingToken()), address(incomingToken));
+    }
+
+    function testSingleCorrectAllocaation() public
+    {
+        // arrange
+        Allocation[] memory daArray = new Allocation[](1);
+        Allocation memory da = Allocation(recipient1, distributor.BASIS_POINTS_MAX());
+
+        daArray[0] = da;
+
+        // act
+        distributor.setAllocations(daArray);
+
+        // assert
+        assertEq(distributor.getAllocationsLength(), 1);
+        assertEq(distributor.getAllocation(0).weight, distributor.BASIS_POINTS_MAX());
+    }   
+
+    function testTwoCorrectAllocation() public
+    {
+        // arrange 
+        Allocation[] memory daArray = new Allocation[](2);
+        Allocation memory da1 = Allocation(recipient1, 8000);
+        Allocation memory da2 = Allocation(recipient2, 2000);
+
+        daArray[0] = da1;
+        daArray[1] = da2;
+
+        // act
+        distributor.setAllocations(daArray);
+
+        // assert
+        assertEq(distributor.getAllocation(0).weight, 8000);
+        assertEq(distributor.getAllocation(1).weight, 2000);
+    }
+
+    function testExpandAllocation() public 
+    {
+        // arrange 
+        Allocation[] memory daArray = new Allocation[](2);
+        Allocation memory da1 = Allocation(recipient1, 6000);
+        Allocation memory da2 = Allocation(recipient2, 4000);
+        daArray[0] = da1;
+        daArray[1] = da2;
```

Comment:
`da1` and `da2` are only referenced once and can be inlined, not 100% sure if solc will compile this but it feels like it should:
```suggestion
        daArray[0] = Allocation(recipient1, 6000);
        daArray[1] = Allocation(recipient1, 4000);
```

Reply:
fixed